lib/spack/spack/fetch_strategy.py

```diff
@@ -95,6 +98,13 @@ def fetcher(cls):
     return cls
 
 
+def _raise_from(new_exc, prev_exc):
+    # type: (BaseException, BaseException) -> None
```

Comment:
Can you add a little docstring here to explain the purpose of this function?

Reply:
Yes! Done.